src/operator/integrate.rs

```diff
@@ -186,4 +270,65 @@ mod test {
             root.step().unwrap();
         }
     }
+
+    #[rustfmt::skip]
```

Comment:
this is not a real plus, given the definition in your program below.
it is a plus with state

Reply:
It's the exact same plus actually.  Why do you say it has state?

Reply:
so there is a z^-1 and a special max(a-1, 0) operator which I didn't notice, which you denoted by "-1".

Reply:
there is no box around the z